src/core.jl

```diff
@@ -4,18 +4,24 @@ abstract AbstractPathState
 _insert_and_dedup!(v::Vector{Int}, x::Int) = isempty(splice!(v, searchsorted(v,x), x))
 
 """A type representing a single edge between two vertices of a graph."""
-typealias Edge Pair{Int,Int}
+immutable Edge
+    src::Int
+    dst::Int
+end
 
-@deprecate rev(e::Edge) reverse(e)
+show(io::IO, e::Edge) = print(io, "Edge $(e.src) => $(e.dst)")
 
 """Return source of an edge."""
-src(e::Edge) = e.first
+src(e::Edge) = e.src
 """Return destination of an edge."""
-dst(e::Edge) = e.second
+dst(e::Edge) = e.dst
+
+srcdst(e::Edge) = (src(e), dst(e))
```

Comment:
Could this be a method of `convert(::Tuple{Int,Int}, e::Edge)` rather than a new function?